sql/src/main/java/io/crate/execution/engine/fetch/NodeFetchOperation.java

```diff
@@ -123,7 +124,11 @@ public NodeFetchOperation(ThreadPoolExecutor executor,
         RootTask context = tasksService.getTask(jobId);
         FetchTask fetchTask = context.getTask(phaseId);
         try {
-            doFetch(fetchTask, resultFuture, docIdsToFetch);
+            doFetch(fetchTask, resultFuture, docIdsToFetch).whenComplete((r, t) -> {
+                if (t != null) {
+                    resultFuture.completeExceptionally(t);
```

Comment:
It seems a bit strange that the `resultFuture` is passed as argument into `doFetch` and completed on the outside in a failure case.

Could we do the failure handling inside of `doFetch`. Otherwise I think the `resultFuture` argument should be removed, and both success and failure cases should be forwarded to the `resultFuture`

Reply:
Good point. Moving the failure handling inside `doFetch`. The success case is handled by the `Collect` tasks once all of them have completed successfully.